pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_schema_change_in_txn failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_schema_change_in_txn [failed](https://tanzanite.cluster.engflow.com/invocations/default/a488de16-6191-4596-88c4-e8aa799c25ea?testReportRun=16&testReportShard=29&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [27468a22f82fb1e500bf7b35a62ca58d80abac39](https://github.com/cockroachdb/cockroach/commits/27468a22f82fb1e500bf7b35a62ca58d80abac39):


```
=== RUN   TestTenantLogic_schema_change_in_txn
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogic_schema_change_in_txn3042879326
    test_log_scope.go:81: use -show-logs to present logs inline
[08:52:31] setting distsql_workmem='62821B';
[08:52:31] setting distsql_workmem='62821B';
    logic.go:4317: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_schema_change_in_txn3042879326
--- FAIL: TestTenantLogic_schema_change_in_txn (11.01s)
=== RUN   TestTenantLogic_schema_change_in_txn/rollback_mutations
[08:52:37] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn: 194 statements
    logic.go:2967: 
         
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn:750: SELECT status,
               regexp_replace(description, 'ROLL BACK JOB \d+.*', 'ROLL BACK JOB') as descr
          FROM [SHOW JOBS] WHERE job_type = 'SCHEMA CHANGE GC' AND description LIKE 'GC for ROLL%' ORDER BY job_id DESC LIMIT 1
        expected:
            running  GC for ROLLBACK of ALTER TABLE test.public.customers ADD CO … ers (n)
            
        but found (query options: "") :
            succeeded  GC for ROLLBACK of ALTER TABLE test.public.customers ADD CO … ers (n)
            
[08:52:38] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn with config 3node-tenant: 195 tests, 1 failures
[08:52:40] --- total progress: 195 statements
--- total: 195 tests, 1 failures
    --- FAIL: TestTenantLogic_schema_change_in_txn/rollback_mutations (5.37s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=16</code>
 - <code>shard=29</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_schema_change_in_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41177

--- a/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn
+++ b/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn
@@ -746,12 +746,12 @@ SELECT status,
 ----
 failed  ALTER TABLE test.public.customers ADD COLUMN i INT8 DEFAULT … ers (n)
 
-query TT
-SELECT status,
+query BT
+SELECT status % '(running)|(succeeded)',
        regexp_replace(description, 'ROLL BACK JOB \d+.*', 'ROLL BACK JOB') as descr
   FROM [SHOW JOBS] WHERE job_type = 'SCHEMA CHANGE GC' AND description LIKE 'GC for ROLL%' ORDER BY job_id DESC LIMIT 1
 ----
-running  GC for ROLLBACK of ALTER TABLE test.public.customers ADD CO … ers (n)
+true  GC for ROLLBACK of ALTER TABLE test.public.customers ADD CO … ers (n)
 
 subtest add_multiple_computed_elements
 

